core/modules/wbase/Task.h

```diff
@@ -50,6 +50,28 @@ namespace lsst {
 namespace qserv {
 namespace wbase {
 
+/** Base class for tracking a database query for a worker Task.
+ */
+class TaskQueryRunner {
+public:
+    using Ptr = std::shared_ptr<TaskQueryRunner>;
+    virtual ~TaskQueryRunner() {};
+    virtual bool runQuery()=0;
+    virtual void cancel()=0;
+};
+
+class Task;
+
+/** Base class for scheduling Tasks.
+ * Allows the scheduler to take appropriate action when a task is cancelled.
+ */
+class TaskScheduler {
```

Comment:
Cool on future.  Re the transplant argument: ya, but doesn't mean you can't take the opportunity to fix/clean e.g. struct vs. class for consistency as you make a pass through the file (like various other minor consistency and style fixes you've done throughout?)